ow_plexil/src/plexil-adapter/joint_support.h

```diff
@@ -7,41 +7,61 @@
 
 // Support for lander joints, based on ROS /joint_states message.
 
+// NOTE: This package (ow_plexil) is implemented for the lander in the
+// ow_simulator package.  Specifically, the joints and joint ordering
+// (in /joint_states) is assumed fixed, and is often hardcoded.
+
 #include <string>
 
-enum class Joint {
-  shoulder_yaw,
-  shoulder_pitch,
-  proximal_pitch,
-  distal_pitch,
-  hand_yaw,
-  scoop_yaw,
-  antenna_pan,
-  antenna_tilt,
-  grinder
+const size_t KnownJoints = 9;
+
+enum Joint {
+  // NOTE: we could use the names in /joint_states, but these seem more readable.
+  ANTENNA_PAN = 0,
+  ANTENNA_TILT = 1,
+  DISTAL_PITCH = 2,
+  GRINDER = 3,
+  HAND_YAW = 4,
+  PROXIMAL_PITCH = 5,
+  SCOOP_YAW = 6,
+  SHOULDER_PITCH = 7,
+  SHOULDER_YAW = 8
 };
 
 struct JointProperties
 {
+  // A structure to store useful static properties about joints.
+
   // Use compiler's default methods.
-  std::string rosName;
   std::string plexilName; // human-readable, no spaces
   double softTorqueLimit;
   double hardTorqueLimit;
 };
 
 struct JointTelemetry
```

Comment:
There is no need for an explicit constructor here in C++14 and beyond. Default values can be given to data members that may then be overwritten with brace-initilization.
```C++
struct JointTelemetry
{
  double position = 0;
  double velocity = 0;
  double effort = 0;
  double acceleration = 0;
}
```

Reply:
How would a construction with specific values be supported, as done in `OwInterface.cpp` : `JointTelemetry (position, velocity, effort)` ?

Reply:
This should work.
```C++
    JointTelemetries[i] = {position, velocity, effort};
```

Reply:
`acceleration` would then just be assigned it's default (0).

Reply:
I went ahead and implemented this and did a quick test.  No problems.  Thanks for the tip!